api/all/src/main/java/io/opentelemetry/api/logs/LogRecordBuilder.java

```diff
@@ -98,9 +103,111 @@ default LogRecordBuilder setAllAttributes(Attributes attributes) {
+  }
+
+  /**
+   * Sets an attribute to the newly created {@code LogRecord}. If {@code LogRecordBuilder}
+   * previously contained a mapping for the key, the old value is replaced by the specified value.
+   *
+   * <p>Note: It is strongly recommended to use {@link #setAttribute(AttributeKey, Object)}, and
+   * pre-allocate your keys, if possible.
+   *
+   * @param key the key for this attribute.
+   * @param value the value for this attribute.
+   * @return this.
+   */
+  default LogRecordBuilder setAttribute(String key, long value) {
+    return setAttribute(AttributeKey.longKey(key), value);
+  }
+
+  /**
+   * Sets an attribute to the newly created {@code LogRecord}. If {@code LogRecordBuilder}
+   * previously contained a mapping for the key, the old value is replaced by the specified value.
+   *
+   * <p>Note: It is strongly recommended to use {@link #setAttribute(AttributeKey, Object)}, and
+   * pre-allocate your keys, if possible.
+   *
+   * @param key the key for this attribute.
+   * @param value the value for this attribute.
+   * @return this.
+   */
+  default LogRecordBuilder setAttribute(String key, double value) {
+    return setAttribute(AttributeKey.doubleKey(key), value);
+  }
+
+  /**
+   * Sets an attribute to the newly created {@code LogRecord}. If {@code LogRecordBuilder}
+   * previously contained a mapping for the key, the old value is replaced by the specified value.
+   *
+   * <p>Note: It is strongly recommended to use {@link #setAttribute(AttributeKey, Object)}, and
+   * pre-allocate your keys, if possible.
+   *
+   * @param key the key for this attribute.
+   * @param value the value for this attribute.
+   * @return this.
+   */
+  default LogRecordBuilder setAttribute(String key, boolean value) {
+    return setAttribute(AttributeKey.booleanKey(key), value);
+  }
+
+  /**
+   * Sets an attribute to the newly created {@code LogRecord}. If {@code LogRecordBuilder}
+   * previously contained a mapping for the key, the old value is replaced by the specified value.
+   *
+   * <p>Note: It is strongly recommended to use {@link #setAttribute(AttributeKey, Object)}, and
+   * pre-allocate your keys, if possible.
+   *
+   * @param key the key for this attribute.
+   * @param value the value for this attribute.
+   * @return this.
+   */
+  default LogRecordBuilder setAttribute(AttributeKey<Long> key, int value) {
+    return setAttribute(key, (long) value);
+  }
+
+  /**
+   * Sets an attribute to the newly created {@code LogRecord}. If {@code LogRecordBuilder}
+   * previously contained a mapping for the key, the old value is replaced by the specified value.
+   *
+   * @param key the key for this attribute.
+   * @param value the value for this attribute.
+   * @return this.
+   */
+  // TODO (trask) how to use pre-encoded key?
+  //  - is ComplexAttributeKey too broad (covers all nested attribute types)?
+  default LogRecordBuilder setAttribute(String key, ComplexAttribute value) {
```

Comment:
Why introduce a new `ComplexAttribute` type instead of just `setAttribute(String, Value)`?

Only reason I can think of is for code generation in `semantic-convention-java`.